fvm/env.go

```diff
@@ -113,6 +116,36 @@ func (e *hostEnv) setTransaction(tx *flow.TransactionBody, txIndex uint32) {
 	)
 }
 
+func (e *hostEnv) GetAuthorizedAccountsForContractUpdatesFunc() func() []common.Address {
+	return func() []common.Address {
+		// set default to service account only
+		service := runtime.Address(e.ctx.Chain.ServiceAddress())
+		defaultAccounts := []runtime.Address{service}
+
+		value, err := e.vm.Runtime.ReadStored(
+			service,
+			cadence.Path{
+				Domain:     blueprints.ContractDeploymentAuthorizedAddressesPathDomain,
+				Identifier: blueprints.ContractDeploymentAuthorizedAddressesPathIdentifier,
+			},
+			runtime.Context{
+				Interface: e,
+				Location:  common.ScriptLocation("temp for now until this becomes optional"),
+			},
+		)
+		if err != nil {
+			e.ctx.Logger.Warn().Msg("failed to read contract deployment authrozied accounts from service account. using default behaviour instead.")
+			return defaultAccounts
+		}
+		adresses, ok := utils.OptionalCadenceValueToAddressSlice(value)
+		if !ok {
+			e.ctx.Logger.Warn().Msg("failed to parse contract deployment authrozied accounts from service account. using default behaviour instead.")
+			return defaultAccounts
+		}
+		return adresses
+	}
+}
```

Comment:
The returned function only closes over the receiver, you can remove the inner function and make the outer function perform the code. Then, instead of calling it (`env.GetAuthorizedAccountsForContractUpdatesFunc()` above), just pass a bound function: `env.GetAuthorizedAccountsForContractUpdates`

Reply:
fixed at c996f15